### PR TITLE
FAL-2482 [LX-2242] 

### PR DESCRIPTION
**Pull request by swalladge**
[FAL-2482](https://tasks.opencraft.com/browse/FAL-2482) | [LX-2242](https://tasks.opencraft.com/browse/LX-2242)

These changes are to support the transition from srt subtitles to inline html transcripts for audio blocks.

**Test instructions**:

- test as part of [opencraft/client/LabXchange/labxchange-dev!1532](https://gitlab.com/opencraft/client/LabXchange/labxchange-dev/-/merge_requests/1532)
- read the code and verify the logic is correct and desirable

### CODE REVIEW

File: labxchange_xblocks/audio_block.py [hunk 1]
```diff
@@ -63,23 +77,33 @@ def student_view_data(self, context=None):
...
                 'language': iso_languages.get(lang),
                 'url': current_asset.url if current_asset else ''
             })
-            if current_asset:
-                transcript = self.get_transcript_content(current_asset.url)
-                sequences = self.get_sequences(transcript)
+
+            # convert legacy srt subtitles on the fly
+            # TODO: save back to self.transcripts
+            # this was throwing an error because couldn't save in LMS and/or without a draft
```
Comment (swalladge) [inline labxchange_xblocks/audio_block.py:91]: @symbolist @pomegranited will this be possible?  If we can convert them on the fly, and save them back to the xblock with the new format, we can quickly phase out and then drop support for srt files here.

Comment (pomegranited) [reply labxchange_xblocks/audio_block.py:92]: @swalladge Sure, (almost) anything's possible.. is that part of the scope of this ticket?  Also, does LX ever want Audio assets to sync with their transcripts? If so, then we need to keep the timestamps provided by SRT, so phasing them out may not be the best plan. Can you check with Tess on this?

But note: we don't want to drop support for importing SRT files, since that's the standard format that we can export from YouTube and other sources.

Comment (swalladge) [reply labxchange_xblocks/audio_block.py:92]: @pomegranited followed up on the ticket

**Review by michaelbukachi — APPROVED** [review]
@swalladge  :+1 looks good

- [x] I tested this: followed test instructions
- [x] I read through the code

File: labxchange_xblocks/audio_block.py [hunk 2]
```diff
@@ -63,23 +78,32 @@ def student_view_data(self, context=None):
...
                 'language': iso_languages.get(lang),
-                'url': current_asset.url if current_asset else ''
             })
-            if current_asset:
-                transcript = self.get_transcript_content(current_asset.url)
-                sequences = self.get_sequences(transcript)
+
+            # Convert legacy srt subtitles on the fly.
+            # These will be permanently saved when the client saves.
+            # TODO: remove this legacy support once all audio blocks on prod have been updated.
```
Comment (pomegranited) [inline labxchange_xblocks/audio_block.py:91]: @swalladge  Could you create an LX ticket for this work and reference its ID here? So we know when/where it's to be done.

Comment (pomegranited) [reply labxchange_xblocks/audio_block.py:91]: And please add a test for this logic -- ok to move to a separate method if that makes it easier to test.

File: labxchange_xblocks/tests/audio_block_test.py [hunk 3]
```diff
@@ -47,15 +46,13 @@ class AudioBlockTestCase(BlockTestCaseBase):
...
-                'sequences': [],
                 'transcripts': {},
                 'user_state': {'current_language': None},
             },
             {
                 'display_name': 'A very cool track',
                 'embed_code': '<iframe width="100%" height="300" scrolling="no" frameborder="no" allow="autoplay" src="https://w.soundcloud.com/player/?url=https%3A//api.soundcloud.com/tracks/794640376&color=%23ff5500&auto_play=false&hide_related=false&show_comments=true&show_user=true&show_reposts=false&show_teaser=true&visual=true"></iframe>',  # noqa: E501
                 'options': [],
-                'sequences': [],
                 'transcripts': {},
```
Comment (pomegranited) [inline labxchange_xblocks/tests/audio_block_test.py:56]: Could you add a test for a case where `transcripts` is not empty?

**Review by pomegranited — CHANGES_REQUESTED** [review]

File: labxchange_xblocks/audio_block.py (hunk 2 above)
Comment (swalladge) [reply labxchange_xblocks/audio_block.py:91]: Created :) https://tasks.opencraft.com/browse/LX-2349

File: labxchange_xblocks/tests/audio_block_test.py (hunk 3 above)
Comment (swalladge) [reply labxchange_xblocks/tests/audio_block_test.py:56]: I can try.  Not sure whether to try mocking the block assets/files, or try testing just that logic.  It's all tied up together... :/

Comment (swalladge) [reply labxchange_xblocks/tests/audio_block_test.py:56]: @pomegranited do you have advice for writing tests here?  I can't find any existing tests that work with assets to base off.

Comment (pomegranited) [reply labxchange_xblocks/tests/audio_block_test.py:56]: How about the tests you added for [Case Study attachments](https://github.com/open-craft/labxchange-xblocks/blob/master/labxchange_xblocks/tests/case_study_block_test.py#L244)?

Comment (swalladge) [reply labxchange_xblocks/tests/audio_block_test.py:56]: @pomegranited nah, case study attachments is just a simple [list field](https://github.com/open-craft/labxchange-xblocks/blob/9f4b7f05badfce562da43d9d8b1887ac81dc6298/labxchange_xblocks/case_study_block.py#L53-L58).

Comment (swalladge) [reply labxchange_xblocks/tests/audio_block_test.py:56]: Working on it.  Will write custom mocks.

Comment (swalladge) [reply labxchange_xblocks/tests/audio_block_test.py:55]: @pomegranited tests added! :)

Comment (pomegranited) [reply labxchange_xblocks/tests/audio_block_test.py:55]: Looks great, thank you @swalladge !
